js/mail.js

```diff
@@ -1194,6 +1185,35 @@ var Mail = {
 $(document).ready(function() {
 	Mail.UI.initializeInterface();
 
+	// Enables infinite scroll ability.
+	setInterval(function() {
+
+		// Element identifies the element we want to trigger our loading on.
+		//var element = $('.mail_message_summary:nth-last-child(1)');
+		var element = $('#load-more-mail-messages');
+		var outerHeight = element.outerHeight();
+		var offset = element.offset();
+		var scrollLength = $('#mail_messages').scrollTop();
+		var screenHeight = $(document).height();
+		var messageListHeight = scrollLength + offset.top + outerHeight;
+
+		if ((scrollLength + screenHeight) >= messageListHeight) {
+			Mail.UI.messageView.loadMessages('addMore');
+		}
+
+	}, 1000);
+
+	/**
+	 * Checks and loads any new messages and changes to the current folder.
+	 * @todo combine this with the checkForNotifications method.
+	 */
+	setInterval((function() {
+		// Begin the loop.
+		return function() {
+			Mail.UI.messageView.loadMessages();
+		};
+	})(), 10*1000);
```

Comment:
Unless you can provide technical feedback, I suggest we leave the interval as it is. Mac Mail and Thunderbird have intervals around this time. I prefer 3 seconds, but I imagine some rare cases where IMAP servers are slow. Tests by @jancborchardt and I revealed that our response time was a fraction of a second. Lets merge this and get some real world results. If anyone reports issues, we can immediately address it then with logic that refreshes based on a finished request as opposed to our time based logic now. (I'd like this function as well, but Jan and I also agree this pull is already quite big.)